internal/services/sources_service.go

```diff
@@ -52,59 +65,95 @@ func GetSource(w http.ResponseWriter, r *http.Request) {
 	resp, err := client.ShowSourceWithResponse(r.Context(), sources.ID(id), AddIdentityHeader)
 	statusCode := resp.StatusCode()
 	if err != nil {
-		renderError(w, r, payloads.SourcesClientError(r.Context(), "show source", err, statusCode))
+		renderError(w, r, payloads.SourcesClientError(r.Context(), "show source", err, statusCode, resp.Body))
```

Comment:
I don't like mixing errors and body in a single argument (variable). Can't we simply create a new struct that will have both? Then you can set one, or the other accordingly.

Reply:
Do you mean to change the signature of `payloads.SourcesClientError `to something like that: 
```golang
func SourcesClientError(ctx context.Context, message string, err error, status int, errors []byte, body []byte) 
```
And pass nil when one or the other is not relevant?